proto/src/main/proto/Rss.proto

```diff
@@ -184,6 +184,7 @@ message ShuffleRegisterRequest {
   string user = 5;
   DataDistribution shuffleDataDistribution = 6;
   int32 maxConcurrencyPerPartitionToWrite = 7;
+  int32 stageAttemptNumber = 8;
```

Comment:
How to reject legacy data?

Reply:
The legacy request could be rejected according to attemptNumber in sendShuffleData + reportShuffleResult rpc.

Reply:
cc @yl09099 Pay more attension.

Reply:
This should be solved.